include/pybind11/pybind11.h

```diff
@@ -216,9 +214,8 @@ class cpp_function : public function {
                 const std::type_info *t = types[type_index++];
                 if (!t)
                     pybind11_fail("Internal error while parsing type signature (1)");
-                auto it = registered_types.find(std::type_index(*t));
```

Comment:
Btw, taking a more detailed look -- this is a completely "cold" piece of code though, likely to only be called once per module initialization? (`cpp_function` setup)